### PR TITLE
net-snmp: Add support for SNMPv3

### PR DESCRIPTION
**Pull request by devran01**
Add openssl as dependency to support SNMPv3 and also create a default
ro/rw users as part of intial configuration. Additionally, include
disman/event MIB to enable linkUpDown notifications.

Signed-off-by: Devaraj Ranganna <devaraj.ranganna@arm.com>

Maintainer: @stintel 
Compile tested: BCM27xx, BCM2709, Raspberry Pi 2 model B, OpenWrt Chaos Calmer 15.05 
Run tested: BCM27xx, BCM2709, Raspberry Pi 2 model B, OpenWrt Chaos Calmer 15.05. Tested with Cacti tool

### CODE REVIEW

File: net/net-snmp/files/snmpd.init [hunk 1]
```diff
@@ -210,6 +210,15 @@ snmpd_engineid_add() {
 	[ -n "$engineidnic" ] && echo "engineIDNic $engineidnic" >> $CONFIGFILE
 }
 
+v3user_add()
```
Comment (pprindeville) [inline net/net-snmp/files/snmpd.init]: Dumb question.... how does this function get invoked?

**Comment by devran01** [comment]
@pprindeville 
Thanks for the review. Fixed with the new commit.

**Comment by stintel** [comment]
Please don't add OpenSSL as a hard dependency. We should probably create an snmpd-openssl variant for this.

**Comment by pprindeville** [comment]
Given the size of net-snmp I'm not too concerned about the size or dependencies.

But given the ability to wreak havoc on a system, I would want to make it as easy as possible to use SNMP securely.

I'd make size subordinate to security.

But that's me: I'm paranoid.

**Comment by stintel** [comment]
After thinking about it for a while, I decided not to accept this PR in its current form. This package has been in the packages feed for years without snmpv3 support. Forcing all existing users to suddenly install openssl does not feel right.

Please create a net-snmp-ssl and net-snmp-nossl variant instead, both with PROVIDES:=net-snmp. It's important that packages name != PROVIDES.

**Comment by omonar** [comment]
How about using --with-openssl=internal flag? It will compile the net-snmp package with stripped down set of crypto routines (des, aes, md5, sha1). See README in snmplib/openssl for details.

**Comment by omonar** [comment]
How about using --with-openssl=internal flag? It would compile the net-snmp package with stripped down set of crypto routines (aes, des, md5, sha1). See README in snmplib/openssl for details.

**Comment by stintel** [comment]
No. In that same README they don't really recommend it. And with all the bugs the are being found in OpenSSL, this is a bad idea for security reasons. Having net-snmp-ssl and net-snmp-nossl is the way to go.

**Comment by champtar** [comment]
Hi @devran01 
you need to rebase and create 2 variant (ssl / nossl), so closing this PR for now

**Pull request closed without merging**